rasa/core/training/generator.py

```diff
@@ -44,9 +44,22 @@ class TrackerWithCachedStates(DialogueStateTracker):
     """A tracker wrapper that caches the state creation of the tracker."""
 
     def __init__(
-        self, sender_id, slots, max_event_history=None, domain=None, is_augmented=False
+        self,
```

Comment:
how about overriding `from_events` in here instead doing changes to the `DialogueStateTracker` which doesn't need the domain?